libs/coroutines/include/hpx/coroutines/detail/context_base.hpp

```diff
@@ -229,6 +244,44 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
 #endif
         }
 
+#ifdef HPX_HAVE_LIBCDS
+        std::size_t get_libcds_data() const
+        {
+            return libcds_data_;
+        }
+
+        std::size_t set_libcds_data(std::size_t data)
+        {
+            static std::size_t olddata = libcds_data_;
```

Comment:
This creates a data race as it is shared amongst threads. Same is true for the other functions below.

Reply:
If in doubt, you could write it as:
```
std::size_t set_libcds_data(std::size_t data)
{
    std::swap(data, libcds_data_);
    return data;
}
```